### PR TITLE
Passport login logout

### PR DESCRIPTION
**Pull request by Saurabh0707**
added auth and passport scaffolding.
Login Logout remaining.

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -4,9 +4,72 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file",
```
Comment (dagasatvik10) [inline composer.lock]: Add this to file to gitignore.
Its not required to be added to the repository.

File: resources/views/auth/login.blade.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
+@extends('layouts.app')
```
Comment (dagasatvik10) [inline resources/views/auth/login.blade.php]: Why is this file needed?
It should be removed.

File: resources/views/auth/passwords/email.blade.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
+@extends('layouts.app')
```
Comment (dagasatvik10) [inline resources/views/auth/passwords/email.blade.php]: This too is not needed.

File: resources/views/auth/passwords/reset.blade.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
+@extends('layouts.app')
```
Comment (dagasatvik10) [inline resources/views/auth/passwords/reset.blade.php]: This also not needed.

File: resources/views/auth/register.blade.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
+@extends('layouts.app')
```
Comment (dagasatvik10) [inline resources/views/auth/register.blade.php]: Same.
Not needed.

File: resources/views/home.blade.php [hunk 6]
```diff
@@ -0,0 +1,23 @@
+@extends('layouts.app')
```
Comment (dagasatvik10) [inline resources/views/home.blade.php]: Not needed.

File: resources/views/layouts/app.blade.php [hunk 7]
```diff
@@ -0,0 +1,80 @@
+<!DOCTYPE html>
```
Comment (dagasatvik10) [inline resources/views/layouts/app.blade.php]: Not needed.

File: routes/web.php [hunk 8]
```diff
@@ -14,3 +14,11 @@
...
     return view('welcome');
 });
+
+Auth::routes();
+
+Route::get('/home', 'HomeController@index')->name('home');
+
+Auth::routes();
+
+Route::get('/home', 'HomeController@index')->name('home');
```
Comment (dagasatvik10) [inline routes/web.php]: Home route is added twice, why? Same goes for Auth::routes().
Also I don't see the need for home route.
You can put a root route(i.e. '/') and return a json their giving API version and Appname.
No need for home route.

File: composer.lock (hunk 1 above)
Comment (Saurabh0707) [reply composer.lock]: okay sir!!!

File: routes/web.php (hunk 8 above)
Comment (Saurabh0707) [reply routes/web.php]: ohkay sir  doing the same.

File: composer.lock (hunk 1 above)
Comment (Saurabh0707) [reply composer.lock]: sir....a doubt!!
it says here that :

Installing Without composer.lock:
     When Composer has finished installing, it writes all of the packages and the exact versions of 
      them that it downloaded to the composer.lock file, locking the project to those specific 
      versions. You should commit the composer.lock file to your project repo so that all people 
      working on the project are locked to the same versions of dependencies (more below).

Installing With composer.lock#

      This brings us to the second scenario. If there is already a composer.lock file as well as a 
      composer.json file when you run composer install, it means either you ran the install 
      command before, or someone else on the project ran the install command and committed 
      the composer.lock file to the project (which is good).

     Either way, running install when a composer.lock file is present resolves and installs all 
     dependencies that you listed in composer.json, but Composer uses the exact versions listed in 
     composer.lock to ensure that the package versions are consistent for everyone working on 
     your project. As a result you will have all dependencies requested by your composer.json file, 
     but they may not all be at the very latest available versions (some of the dependencies listed in 
     the composer.lock file may have released newer versions since the file was created). This is by 
     design, it ensures that your project does not break because of unexpected changes in 
     dependencies.

Commit Your composer.lock File to Version Control#

     Committing this file to VC is important because it will cause anyone who sets up the project to 
     use the exact same versions of the dependencies that you are using. Your CI server, production 
     machines, other developers in your team, everything and everyone runs on the same 
    dependencies, which mitigates the potential for bugs affecting only some parts of the 
    deployments. Even if you develop alone, in six months when reinstalling the project you can 
    feel confident the dependencies installed are still working even if your dependencies released 
    many new versions since then. (See note below about using the update command.)

So according to this we must commit composer.lock to the repo so that all other contributors remain consistent with my version of any package i use....!!!!

Comment (Saurabh0707) [reply composer.lock]: so what do i do?

Comment (Saurabh0707) [reply composer.lock]: @dagasatvik10 ....sir?

Comment (Saurabh0707) [reply composer.lock]: If I would update my libs, then I have to commit the lockfile too. It basically states that my project is locked to those specific versions of the libs I m using.

If I commit my changes, and someone pulls my code and updates the dependencies, the lockfile should be unmodified. If it is modified, it means that I have a new version of something.

Having it in the repository assures me that each developer is using the same versions.

Comment (Saurabh0707) [reply composer.lock]: @dagasatvik10 ...sir should i commit it?

**Comment by dagasatvik10** [comment]
Merged

File: composer.lock (hunk 1 above)
Comment (Saurabh0707) [reply composer.lock]: @AkshatSrivastava2  delete composer.lock before merging this PR with your fork. It needs to remain consistent with all contributors.
